src/test/java/org/folio/rest/unit/ItemDamagedStatusAPIUnitTest.java

```diff
@@ -0,0 +1,333 @@
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(500, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void deleteItemDamagedStatusesByIdShouldReturnInternalServerError(TestContext testContext) {
+    doThrow(new RuntimeException(INTERNAL_SERVER_ERROR_MSG))
+      .when(postgresClient)
+      .delete(
+        eq(REFERENCE_TABLE),
+        anyString(),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.deleteItemDamagedStatusesById(
+      UUID.randomUUID().toString(),
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(500, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void deleteItemDamagedStatusesByIdShouldReturnBadRequest(TestContext testContext) {
+    GenericDatabaseException exception = new GenericDatabaseException(new ErrorMessage(getUUIDErrorMap()));
+    doAnswer(setExceptionForHandlerArgument(2, exception))
+      .when(postgresClient)
+      .delete(
+        eq(REFERENCE_TABLE),
+        anyString(),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.deleteItemDamagedStatusesById(
+      UUID.randomUUID().toString(),
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(400, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void putItemDamagedStatusesByIdShouldReturnBadRequest(TestContext testContext) {
+    GenericDatabaseException exception = new GenericDatabaseException(new ErrorMessage(getUUIDErrorMap()));
+    doAnswer(setExceptionForHandlerArgument(3, exception))
+      .when(postgresClient)
+      .update(
+        eq(REFERENCE_TABLE),
+        any(ItemDamageStatus.class),
+        anyString(),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.putItemDamagedStatusesById(
+      UUID.randomUUID().toString(),
+      DEFAULT_LANGUAGE,
+      new ItemDamageStatus(),
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(400, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void putItemDamagedStatusesByIdShouldInternalServerError(TestContext testContext) {
```

Comment:
Under what circumstances should a PUT result in a server error?

Reply:
a developer mistake

Reply:
What kind of developer mistake, one that causes a specific kind of exception to occur?

Reply:
It's standard catch if something goes wrong, during refactoring or something else an exception will be wrapped in 500 response